database_test.go

```diff
@@ -67,7 +67,14 @@ func (ds *databaseSuite) TestScanStructs() {
 		WithArgs().
 		WillReturnRows(sqlmock.NewRows([]string{"address", "name"}).
 			FromCSVString("111 Test Addr,Test1\n211 Test Addr,Test2"))
-
+	mock.ExpectQuery(`SELECT \* FROM "items"`).
```

Comment:
Why were these additional expects added here and below?

Reply:
Now targets validation is done right in `ScanRows()`, after query is executed by db, thus error mock should expect query before error will be returned.

As I wrote above, this should not be a problem because invalid parameters for scan rows is a kind of error that should be detected by basic tests at early stages of development. So in production this error is not likely to occur.

Of course there is an alternative that keeps previous behaviour, but is seem to be ugly, like need some flag to tell scanner that validation was already done.

Reply:
Got it, looking at https://github.com/doug-martin/goqu/blob/eda0d817d80bd64ad75c2aa0151b717d0edb492b/exec/query_executor.go#L241:24 it makes sense.